src/Database/Relations/AttachOneOrMany.php

```diff
@@ -39,7 +39,14 @@ public function isPublic()
     public function addConstraints()
     {
         if (static::$constraints) {
-            $this->query->where($this->morphType, $this->morphClass);
+            if (starts_with($this->morphClass, 'Winter\\')) {
+                $this->query->where(function ($q) {
+                    $q->where($this->morphType, $this->morphClass);
+                    $q->orWhere($this->morphType, str_replace('Winter\\', 'RainLab\\', $this->morphClass));
+                });
+            } else {
+                $this->query->where($this->morphType, $this->morphClass);
+            }
```

Comment:
I'm not comfortable with having this in the core, it's pretty hacky to have a vendor-specific conditional in the core and there could be a number of unforeseen issues with invisibly grabbing more data than expected by the developer.

Reply:
isn't that what you meant by:

NOTE: We could also explore the idea of having Laravel replace these values at the query level seeing as it's highly likely that other plugins could be extending the replaced plugins and referencing those models in their own data structure.

Reply:
I more so meant doing so at the plugin level through an API of some sort, but wasn't quite sure how it'd work or if it'd even be feasible.